routing/base/astar_algorithm.hpp

```diff
@@ -12,6 +12,13 @@
 namespace routing
 {
 
+template <typename TVertexType>
+struct RoutingResult
+{
+  vector<TVertexType> path;
```

Comment:
For me it's OK, but note that this may lead to a hard-to-debug bugs, when your ctors'll become more complex. Note that @deathbaba won't be happy with such field names :)